rfd/0152-automatic-database-users-mongodb.md

```diff
@@ -0,0 +1,202 @@
+---
+authors: STeve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+# RFD 152 - Database Automatic User Provisioning for MongoDB
+
+## Required approvers
+
+Engineering: @r0mant || @smallinsky
+Product: @klizhentas || @xinding33
+Security: @reedloden || @jentfoo
+
+## What
+
+This RFD discusses on how to expand Database Automatic Provisioning feature for
+MongoDB.
+
+## Why
+
+Automatic User Provisioning has been implemented for several SQL databases,
+including PostgreSQL and MySQL, with the basic design described in [RFD
+113](https://github.com/gravitational/teleport/blob/master/rfd/0113-automatic-database-users.md).
+
+Adding support for database user provisioning to MongoDB presents unique
+challenges due to differences in architecture compared to traditional SQL
+databases.
+
+This RFD aims to identify the challenges and provide solutions to address them.
+
+Since the differences in architecture between MongoDB Atlas and self-hosted
+MongoDB are significant, they will be discussed separately in this RFD.
+
+## MongoDB Atlas Details
+
+Automatic User Provisioning will NOT be supported for MongoDB Atlas with the
```

Comment:
An additional concern with MongoDB Atlas would be needing to handle scheduled triggers (if users are able to create them)

Reply:
Based on the document, scheduled triggers are created by "Atlas users", not "database users" so we should be good.

This does bring another topic on using Atlas SDK. In order to use the SDK, you have to create an API key pair, and the public key is basically the "Atlas user". Teleport must have both public and private key to authenticate as this "Atlas user". I don't think it's a good idea to hardcode the private key in our config or db definition right? 

In my prototype, i was acquiring pub/private keys through env vars (similar to AWS_xxxx env vars). Is that OK to do from the security point of view?

A related usage problem is we have to figure out how to specify multiple pub/private keypairs for different Atlas projects. 

https://www.mongodb.com/docs/atlas/configure-api-access/